modules/client/test/unit/client/utils.test.ts

```diff
@@ -0,0 +1,1488 @@
+        [upgradeToAndCallAction],
+        daoAddress,
+        pspAddress,
+        currentDaoVersion,
+      );
+      expect(result.isValid).toEqual(false);
+      expect(result.causes).toMatchObject([
+        DaoUpdateProposalInvalidityCause
+          .INVALID_UPGRADE_TO_AND_CALL_VERSION,
+      ]);
+    });
+    it("should return INVALID_UPGRADE_TO_AND_CALL_DATA when the data in the encoded initializeFrom action is not empty", () => {
+      initializeFromAction = client.encoding.initializeFromAction(
+        daoAddress,
+        {
+          previousVersion: currentDaoVersion,
+          initData: new Uint8Array([0, 10, 20, 30]),
+        },
+      );
+      upgradeToAndCallAction = client.encoding.upgradeToAndCallAction(
+        daoAddress,
+        {
+          implementationAddress,
+          data: initializeFromAction.data,
+        },
+      );
+      const result = validateUpdateDaoProposalActions(
+        [upgradeToAndCallAction],
+        daoAddress,
+        pspAddress,
+        currentDaoVersion,
+      );
+      expect(result.isValid).toEqual(false);
+      expect(result.causes).toMatchObject([
+        DaoUpdateProposalInvalidityCause
+          .INVALID_UPGRADE_TO_AND_CALL_DATA,
+      ]);
+    });
+  });
+  describe("isDaoUpdateAction", () => {
+    it("should return the expected output given a specific input", () => {
+      const cases = [
+        { input: [ProposalActionTypes.UPGRADE_TO], expected: true },
+        { input: [ProposalActionTypes.UPGRADE_TO_AND_CALL], expected: true },
+        {
+          input: [
+            ProposalActionTypes.UPGRADE_TO,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.UPGRADE_TO_AND_CALL,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: false,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+            ProposalActionTypes.UPGRADE_TO,
+          ],
+          expected: false,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
```

Comment:
will this check fail if we put in the an update block that does require ROOT_PERM?

Reply:
This check will fail if there is no UPGRADE_TO or UPGRADE_TO_AND_CALL function or if its not on the first position

Reply:
What I wanted to see is you mix the different update action block types (one that does and one that does not require ROOT, e.g.,
```
          input: [
            ProposalActionTypes.UPGRADE_TO,
            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
            ProposalActionTypes.APPLY_UPDATE,
            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
            //
            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
            ProposalActionTypes.GRANT_ROOT_PERMISSION,
            ProposalActionTypes.APPLY_UPDATE,
            ProposalActionTypes.REVOKE_ROOT_PERMISSION,
            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
          ],
```